library/src/visualize/draw.rs

```diff
@@ -0,0 +1,521 @@
+    /// different values.
+    #[default(RadiusValue::Single(Abs::cm(1.0).into()))]
+    r: RadiusValue,
+    /// The angle to rotate the ellipse. Is only relevant if `r` is a two-element array with
+    /// different values.
+    #[named]
+    #[default]
+    x_rotation: Angle,
+    /// Whether the arc should follow the path that is greater than 180 degrees/
+    #[named]
+    #[default]
+    large: bool,
+    /// Whether the arc should start in a positive or negative angle.
+    #[named]
+    #[default]
+    sweep: bool,
+) -> Value {
+    if x.is_some() && dx.v.is_some() {
+        bail!(dx.span, "cannot have both a `x` and `dx` argument")
+    }
+    if y.is_some() && dy.v.is_some() {
+        bail!(dy.span, "cannot have both a `y` and `dy` argument")
+    }
+
+    let x_val = x
+        .map(DeltaAbs::Abs)
+        .or(dx.v.map(DeltaAbs::Delta))
+        .unwrap_or(DeltaAbs::Delta(Rel::zero()));
+    let y_val = y
+        .map(DeltaAbs::Abs)
+        .or(dy.v.map(DeltaAbs::Delta))
+        .unwrap_or(DeltaAbs::Delta(Rel::zero()));
+
+    let mut path = PathBuilder::new();
+    path.arc_to(Axes::new(x_val, y_val), r.to_axes(), x_rotation, large, sweep);
+    path.into()
+}
+
+#[derive(Debug, Clone, Hash)]
+pub enum RadiusValue {
+    Single(Length),
+    Axes(Axes<Length>),
+}
+
+impl RadiusValue {
+    fn to_axes(self) -> Axes<Length> {
+        match self {
+            Self::Single(v) => Axes::splat(v),
+            Self::Axes(a) => a,
+        }
+    }
+}
+
+cast_from_value! {
+    RadiusValue,
+    v: Length => Self::Single(v),
+    a: Axes<Length> => Self::Axes(a),
+}
+
+/// Creates a path segment containing a single closed circle.
+///
+/// ## Example
+/// ```example
+/// #draw.path(stroke: blue, fill: blue.lighten(80%) closed: true,
+///   draw.circle(x: 30pt, y: 30pt, r: 15pt))
+/// ```
+///
+/// Display: Circle
+/// Category: visualize
+/// Returns: path
+#[func]
+pub fn circle(
+    /// The x coordinate of the circle center.
+    #[named]
+    x: Option<Rel<Length>>,
+    /// The y coordinate of the circle center.
+    #[named]
+    y: Option<Rel<Length>>,
+    /// The delta x coordinate of the circle center. Cannot be used together with `x`.
+    #[named]
+    #[default(Spanned::new(None, Span::detached()))]
+    dx: Spanned<Option<Rel<Length>>>,
+    /// The delta y coordinate of the circle center. Cannot be used together with `y`.
+    #[named]
+    #[default(Spanned::new(None, Span::detached()))]
+    dy: Spanned<Option<Rel<Length>>>,
+    /// The radius of the circle
+    #[named]
+    #[default(Abs::cm(1.0).into())]
+    r: Length,
+    /// Whether to draw the circle path in ccw direction, this can be used to create holes in
+    /// a shape
+    /// ```example
+    /// #draw.path(stroke: blue, fill: blue.lighten(80%), {
+    ///   draw.moveto(x: 20pt, y: 20pt)
+    ///   draw.circle(r: 20pt)
+    ///   draw.circle(r: 10pt, ccw: true)
+    /// })
+    /// ```
+    #[named]
+    #[default]
+    ccw: bool,
```

Comment:
How about renaming `ccw` to `counterclockwise`? (or, even shorter, invert the logic and name it `clockwise`)

Reply:
Arcs already have a `sweep` parameter which is the same thing (taken from [svg](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#arcs)). They should probably at least be consistent. I don't know if `sweep` is much clearer though? `counterclockwise` and `clockwise` are at least both quite clear, but maybe a bit verbose?